### PR TITLE
Add options to refresh icon of asset on frontend

### LINKED ISSUE #4243: Add option to refresh icons
## Abstract

Add the possibility when editing an asset to request the deletion of the icon and re-query it

## Motivation

Some projects have updated their token logos and if you don't manually delete the copy of the logo locally you can't query the new one

## Specification

For this a new endpoint should be added that deletes the local copy of the image for the logo and downloads a new one using the current available methods

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #4243

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

image

### CODE REVIEW

**Comment by yabirgb** [comment]
So cute the icon :yum:

File: frontend/app/src/locales/en.json [hunk 1]
```diff
@@ -934,6 +934,10 @@
       "title": "Icon upload",
       "description": "The icon upload failed: {message}"
     },
+    "refresh_icon": {
+      "title": "Refresh icon",
```
Comment (yabirgb) [inline frontend/app/src/locales/en.json:938]: Maybe this could be something that explains better the functionality? What about `Restore icon`?

File: frontend/app/src/components/helper/display/icons/AssetIcon.vue [hunk 2]
```diff
@@ -99,7 +100,8 @@ export default defineComponent({
         return `/assets/images/defi/weth.svg`;
       }
       const url = `${api.serverUrl}/api/1/assets/${id}/icon`;
-      return get(changeable) ? `${url}?t=${Date.now()}` : url;
+      const currentTimestamp = get(timestamp) || Date.now();
```
Comment (yabirgb) [inline frontend/app/src/components/helper/display/icons/AssetIcon.vue:103]: What is the advantage of this vs just using `Date.now()`?

**Review by yabirgb — COMMENTED** [review]
Functionality looks okey. Just one question + one small thing to make it clear. Also could you update the docs for the functionality?

File: frontend/app/src/components/helper/display/icons/AssetIcon.vue (hunk 2 above)
Comment (lukicenturi) [reply frontend/app/src/components/helper/display/icons/AssetIcon.vue:103]: Date.now() is used as a random string to append to the URL.
This is to get the latest image.
get(timestamp) here is used to send some kind of signal from parent component (when icon is refreshed) to tell this component to refetch the image.

File: frontend/app/src/locales/en.json (hunk 1 above)
Comment (lukicenturi) [reply frontend/app/src/locales/en.json:938]: `restore` seems not to be the right word. 
What about `Get the latest icon`?

Comment (yabirgb) [reply frontend/app/src/locales/en.json:938]: That sounds better :)

**Comment by lukicenturi** [comment]
> Functionality looks okey. Just one question + one small thing to make it clear. Also could you update the docs for the functionality?

Added

**Review by yabirgb — COMMENTED** [review]
The deletion of `.vue` files is intended?
Also what are those files being deleted in the `.github` folder?
Finally the other day Lefteris told me to stick `.png` for the images, can you change it to `.png` instead of `.jpg`?

**Comment by lukicenturi** [comment]
> The deletion of `.vue` files is intended? Also what are those files being deleted in the `.github` folder? Finally the other day Lefteris told me to stick `.png` for the images, can you change it to `.png` instead of `.jpg`?

It was mistake.
It's updated to png.

File: docs/usage_guide.rst [hunk 3]
```diff
@@ -1092,7 +1092,9 @@ You can fill in the following fields:
...
 6. Cryptocompare identifier. This is optional but recommended. At least one of coingecko or cryptocompare should be given so that prices can be queried. If not given, the symbol of the asset will be used. If that fails, then cryptocompare is not used. To get the cryptocompare identifier, search for the coin in cryptocompare, visit its url and take it from there. For example for https://www.cryptocompare.com/coins/eth/overview/USD the identifier is ``ETH``. It's always what comes after ``coins``.
-7. You can upload an icon for the asset. Any of the common image extensions is accepted (png, jpg, jpeg, webp). The custom icon always takes precedence over the one auto-detected by coingecko.
+7. Optional fields section. This section will be explained below.
```
[the remark this answers is not in the document]
Comment (lukicenturi) [reply docs/usage_guide.rst:1095]: This number 1 - 9 is shown in the image above. Previously it was a mismatch.

File: docs/usage_guide.rst [hunk 4]
```diff
@@ -1078,7 +1078,7 @@ You can inspect the list of all supported assets, edit them, delete them or add
 Adding/editing an asset
 =======================
 
-.. image:: images/rotki_add_edit_token.png
```
Comment (LefterisJP) [inline docs/usage_guide.rst:1081]: This is still jpg. You need to undo this change.

File: frontend/app/components.d.ts [hunk 5]
```diff
@@ -1,369 +0,0 @@
-// generated by unplugin-vue-components
```
Comment (LefterisJP) [inline frontend/app/components.d.ts:1]: wouldn't this be just removed when we merge bugfixes to develop?

**Review by LefterisJP — COMMENTED** [review]
Two questions. Will leave code lookup to kelsos too.

**Review by kelsos — APPROVED** [review]
lgtm thank you @lukicenturi
